### PR TITLE
fix disengage

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
The platform might disengage prematurely when platform is close to finishing the route and stopping. Control plugins currently continuously publish commands despite being disengaged using the last trajectory available. This PR lets the control plugins reset the trajectory if the guidance is disengaged. 
We may not have detected the behavior previously in real life because the vehicle either successfully commanded 0 before disengaging by chance, or since real life vehicles change the gear to park when disengaged. 

## Related GitHub Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2280

## Related Jira Key
TODO

## Motivation and Context
full integration testing cdar vru use case 1

## How Has This Been Tested?
simulation pc 1 with cdasim

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: carma_guidance_plugins/include/carma_guidance_plugins/control_plugin.hpp [hunk 1]
```diff
@@ -54,7 +55,7 @@ namespace carma_guidance_plugins
     void current_pose_callback(geometry_msgs::msg::PoseStamped::UniquePtr msg);
     void current_twist_callback(geometry_msgs::msg::TwistStamped::UniquePtr msg);
     void current_trajectory_callback(carma_planning_msgs::msg::TrajectoryPlan::UniquePtr msg);
-
+    void guidance_state_callback(carma_planning_msgs::msg::GuidanceState::UniquePtr plan);
```
Comment (JonSmet) [inline carma_guidance_plugins/include/carma_guidance_plugins/control_plugin.hpp:58]: Can we change `plan` here to `msg`?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: carma_guidance_plugins/include/carma_guidance_plugins/control_plugin.hpp (hunk 1 above)
Comment (MishkaMN) [reply carma_guidance_plugins/include/carma_guidance_plugins/control_plugin.hpp]: renamed

**Comment by MishkaMN** [comment]
Please hold off on merging

**Comment by MishkaMN** [comment]
Closing this PR due to prioritization. Created an issue related to it and will pick up later if time allows

**Pull request closed without merging**